pkg/detectors/uri/uri.go

```diff
@@ -19,85 +24,113 @@ type Scanner struct {
 }
 
 // Ensure the Scanner satisfies the interface at compile time.
-var _ detectors.Detector = (*Scanner)(nil)
-var _ detectors.CustomFalsePositiveChecker = (*Scanner)(nil)
+var _ interface {
+	detectors.Detector
+	detectors.CustomFalsePositiveChecker
+} = (*Scanner)(nil)
 
 var (
 	keyPat = regexp.MustCompile(`\b(?:https?:)?\/\/[\S]{3,50}:([\S]{3,50})@[-.%\w\/:]+\b`)
 
 	// TODO: make local addr opt-out
 	defaultClient = detectors.DetectorHttpClientWithNoLocalAddresses
+
+	hostNotFoundCache = simple.NewCache[struct{}]()
 )
 
 // Keywords are used for efficiently pre-filtering chunks.
 // Use identifiers in the secret preferably, or the provider name.
 func (s Scanner) Keywords() []string {
-	return []string{"http"}
+	return []string{"http://", "https://"}
```

Comment:
As far as I can tell, `http://` and `https://` are more correct and efficient keywords.